lib/Net/Async/Blockchain/BTC.pm

```diff
@@ -123,9 +125,18 @@ sub subscribe {
     $subscription = $subscription_dictionary{$subscription};
 
     die "Invalid or not implemented subscription" unless $subscription && $self->can($subscription);
+    my ($zmq_client_source, $zmq_client_socket) = $self->new_zmq_client->subscribe($subscription);
+
+    my $error_handler = sub {
+        my $error = shift;
+        $self->source->fail($error);
+        zmq_close($zmq_client_socket);
+    };
+
+## Please see file lib/Net/Async/Blockchain/BTC.pm.ERR
     Future->needs_all(
-        $self->new_zmq_client->subscribe($subscription)->map(async sub { await $self->$subscription(shift) })->ordered_futures->completed(),
-        $self->recursive_search());
+        $zmq_client_source->map(async sub { await $self->$subscription(shift) })->ordered_futures->completed(),
```

Comment:
Seems a little bit suspicious, we handle failure here, but where do we handle success?

Reply:
We process the received data from the source in the `map` which will do some operations on this data then emit the result to another source.